005_hackerrank_challenges/001_introduction/006_filter_positions_in_a_list/solution.hs

```diff
@@ -1,11 +1,40 @@
+#! /usr/bin/env nix-shell
+#! nix-shell ../../../004_an_updated_nix_shell_file/shell.nix  -i runghc
+
+-- run it locally by doing `cat sample-input.txt | ./solution.hs`
+
+import Data.Maybe (catMaybes)
+import Control.Monad
+
+-- I actually like this solution its very simple
+-- and based solely on the recursive structure of
+-- the list and does exactly what the problem statement
+-- requires
 f :: [Int] -> [Int]
-f lst = go lst
-    where go [] = []
-          go [x] = []
-          go [x,y] = [y]
-          go (x:y:zs) = y:go zs
+f [] = []
+f [_] = []
+f [_,y] = [y]
+f (_:y:zs) = y:f zs
```

Comment:
Since your inner helper function `go` is the exact same as the outer function you can just drop it. That is basically the change here